itl/command-plugins.conf

```diff
@@ -557,6 +557,206 @@ object CheckCommand "http" {
+		}
+		"-l" = {
+			value = "$curl_linespan$"
+			description = "Allow regex to span newlines (must precede -r or -R)"
+		}
+		"-r" = {
+			value = "$curl_ereg$"
+			description = "Search page for regex STRING"
+		}
+		"-R" = {
+			value = "$curl_eregi$"
+			description = "Search page for case-insensitive regex STRING"
+		}
+		"--invert-regex" = {
+			set_if = "$curl_invert_regex$"
+			description = "When using regex, return CRITICAL if found, OK if not"
+		}
+		"--state-regex" = {
+			value = "$curl_state_regex$"
+			description = "Return STATE if regex is found, OK if not"
+		}
+		"-a" = {
+			value = "$curl_authorization$"
+			description = "Username:password on sites with basic authentication"
+		}
+		"-b" = {
+			value = "$curl_proxy_authorization$"
+			description = "Username:password on proxy-servers with basic authentication"
+		}
+		"-A" = {
+			value = "$curl_user_agent$"
+			description = "String to be sent in http header as 'User Agent'"
+		}
+		"-k" = {
+			value = "$curl_header$"
+			repeat_key = true
+			description = "Any other tags to be sent in http header. Use multiple times for additional headers"
+		}
+		"-E" = {
+			set_if = "$curl_extended_perfdata$"
+			description = "Print additional performance data"
+		}
+		"-B" = {
+			set_if = "$curl_show_body$"
+			description = "Print body content below status line"
+		}
+		"-L" = {
+			set_if = "$curl_link$"
+			description = "Wrap output in HTML link (obsoleted by urlize)"
+		}
+		"-f" = {
+			value = "$curl_onredirect$"
+			description = "Options: <ok|warning|critical|follow|sticky|stickyport|curl> How to handle redirected pages."
+		}
+		"--max-redirs" = {
+			value = "$curl_max_redirs$"
+			description = "Maximal number of redirects (default: 15)"
+		}
+		"-m" = {
+			value = "$curl_pagesize$"
+			description = "Minimum page size required (bytes) : Maximum page size required (bytes)"
+		}
+		"--http-version" = {
+			value = "$curl_http_version$"
+			description = "Connect via specific HTTP protocol. 1.0 = HTTP/1.0, 1.1 = HTTP/1.1, 2.0 = HTTP/2 (HTTP/2 will fail without -S)"
+		}
+		"--enable-automatic-decompression" = {
+			set_if = "$curl_enable_automatic_decompression$"
+			description = "Enable automatic decompression of body (CURLOPT_ACCEPT_ENCODING)."
+		}
+		"--haproxy-protocol" = {
+			set_if = "$curl_haproxy_protocol$"
+			description = "Send HAProxy proxy protocol v1 header (CURLOPT_HAPROXYPROTOCOL)"
+		}
+		"--cookie-jar" = {
+			value = "$curl_cookie_jar_file$"
+			description = "Store cookies in the cookie jar file and send them out when requested."
+		}
+		"-w" = {
+			value = "$curl_warning$"
+			description = "Response time to result in warning status (seconds)"
+		}
+		"-c" = {
+			value = "$curl_critical$"
+			description = "Response time to result in critical status (seconds)"
+		}
+		"-t" = {
+			value = "$curl_timeout$"
+			description = "Seconds before connection times out (default: 10)"
+		}
+	}
+
+	vars.curl_ip = "$check_address$"
```

Comment:
As you're extending `ipv4-or-ipv6`, you might wanna set `vars.check_ipv4` and `vars.check_ipv6` as well based on `$curl_ipv4$` and `$curl_ipv6$`. For example:

https://github.com/Icinga/icinga2/blob/894d6aa290e83797d001fcc2887611b23707dbf9/itl/command-plugins.conf#L265-L266

Reply:
done